directive/authorized.go

```diff
@@ -55,102 +56,114 @@ func extractNestedKeyFromArgs(args map[string]any, paramName string) (string, er
 }
 
 func Authorized(openfgaClient openfgav1.OpenFGAServiceClient, log *logger.Logger) func(context.Context, interface{}, graphql.Resolver, string, *string, *string, string) (interface{}, error) {
+	compLogger := log.ComponentLogger("authorizedDirective")
+
 	if !directiveConfiguration.DirectivesAuthorizationEnabled {
+		log.Trace().Msg("Authorization directive is disabled. Skipping authorization check.")
 		return func(ctx context.Context, obj interface{}, next graphql.Resolver, relation string, entityType *string, entityTypeParamName *string, entityParamName string) (interface{}, error) {
 			return next(ctx)
 		}
 	}
 
 	return func(ctx context.Context, obj interface{}, next graphql.Resolver, relation string, entityType *string, entityTypeParamName *string, entityParamName string) (interface{}, error) {
-
-		if openfgaClient == nil {
-			return nil, errors.New("OpenFGAServiceClient is nil. Cannot process request")
-		}
-
-		ctx, err := setTenantToContextForTechnicalUsers(ctx, log)
+		check, err := executeTheAuthCheck(ctx, openfgaClient, compLogger, entityParamName, entityTypeParamName, entityType, relation, next)
 		if err != nil {
-			return nil, err
+			compLogger.Error().Err(err).Msg("error in authorized directive")
+			return check, sentry.SentryError(err)
 		}
 
-		token, err := openmfpcontext.GetAuthHeaderFromContext(ctx)
-		hasToken := err == nil
+		return check, err
```

Comment:
I would explicitly do `return check, nil` as the error is already checked and this is easier to read